pymc/distributions/continuous.py

```diff
@@ -229,32 +229,18 @@ def get_tau_sigma(tau=None, sigma=None):
     -----
     If neither tau nor sigma is provided, returns (1., 1.)
     """
-    if tau is None:
-        if sigma is None:
-            sigma = 1.0
-            tau = 1.0
-        else:
-            if isinstance(sigma, Variable):
-                # Keep tau negative, if sigma was negative, so that it will fail when used
-                tau = (sigma**-2.0) * pt.sign(sigma)
-            else:
-                sigma_ = np.asarray(sigma)
-                if np.any(sigma_ <= 0):
-                    raise ValueError("sigma must be positive")
-                tau = sigma_**-2.0
+    if tau is not None and sigma is not None:
+        raise ValueError("Can't pass both tau and sigma")
 
+    if tau is None and sigma is None:
+        sigma = pt.as_tensor_variable(1.0)
+        tau = pt.as_tensor_variable(1.0)
+    elif tau is None:
+        sigma = pt.as_tensor_variable(sigma)
+        tau = (sigma**-2.0) * pt.sign(sigma)
     else:
-        if sigma is not None:
-            raise ValueError("Can't pass both tau and sigma")
-        else:
-            if isinstance(tau, Variable):
-                # Keep sigma negative, if tau was negative, so that it will fail when used
-                sigma = pt.abs(tau) ** (-0.5) * pt.sign(tau)
-            else:
-                tau_ = np.asarray(tau)
-                if np.any(tau_ <= 0):
-                    raise ValueError("tau must be positive")
-                sigma = tau_**-0.5
+        tau = pt.as_tensor_variable(tau)
+        sigma = pt.abs(tau) ** (-0.5) * pt.sign(tau)
```

Comment:
This actually doesn't need the sign trick, because it will `nan` if tau is negative which is fine enough

Reply:
If we're going to keep the behavior that the returned tau is negative if the passed sigma is negative, then I think we should ensure that the same is true when the user passes negative tau: we should return a negative sigma and it will fail when used.

Reply:
The reason for the abs was just not to mask invalid parameters. The -2 power would make a negative float positive without any complaints. 

That is not a concern with the -0.5 power, which will be nan.

Reply:
The reason this matters is that we get a simpler graph still without silent erros

Reply:
If anything depended on this feel free to revert the changes and we merge your previous iteration